llvm/lib/Transforms/InstCombine/InstCombineSimplifyDemanded.cpp

```diff
@@ -1740,9 +1740,13 @@ Value *InstCombinerImpl::SimplifyDemandedVectorElts(Value *V,
         Constant *CElt = CV->getAggregateElement(i);
         if (isa<ConstantExpr>(CElt))
           continue;
+
         // TODO: If a select condition element is undef, we can demand from
         // either side. If one side is known undef, choosing that side would
         // propagate undef.
+        if (isa<UndefValue>(CElt))
+          continue;
```

Comment:
Instead of explicitly checking for `isa<UndefValue>`, it would be better to check `isOneValue()` for the else below. That way we do not rely on a full enumeration of possibilities. (We can also drop the ConstantExpr check above in that case.)